lib/audited/auditor.rb

```diff
@@ -319,17 +310,28 @@ def auditing_enabled=(val)
         Audited.store["#{table_name}_auditing_enabled"] = val
       end
 
-      protected
       def default_ignored_attributes
-        [primary_key, inheritance_column] + Audited.ignored_attributes
+        [primary_key, inheritance_column] | Audited.ignored_attributes
       end
 
+      protected
+
       def normalize_audited_options
         audited_options[:on] = Array.wrap(audited_options[:on])
         audited_options[:on] = [:create, :update, :destroy] if audited_options[:on].empty?
         audited_options[:only] = Array.wrap(audited_options[:only]).map(&:to_s)
         audited_options[:except] = Array.wrap(audited_options[:except]).map(&:to_s)
       end
+
+      def calculate_non_audited_columns
```

Comment:
So I changed this calculation up a bit. Mostly to drive home the point that the `default_ignored_attributes` should always be included. Previously, they were only considered for the `except` code path and so `only` vs `except` had inconsistent construction logic for deciding which columns to ignore. Here are some other notes I wrote to myself in comments on this: 

```rb
# since there are technically three states
# i.e. audit _only_ these columns, or _except_ these columns
# or _all_ columns (except default ignored), we should probably be explicit about this
# the previously written code was not and it was pretty hard to follow
# and track down bugs. This proposed "new" way has the benefit of being
# more explicit and thus the intent is more obvious
# also the way it was written before broke the rspec matchers since
# in the previous way there would be extra columns include for
# the `except` option because of the original `default_ignored_attributes`
```